src/components/dashboard/widget/dashboard-widget.component.ts

```diff
@@ -42,10 +42,32 @@ export class DashboardWidgetComponent implements OnInit, AfterViewInit, OnDestro
     }
 
     /** Defines the number of columns this widget should occupy. */
-    @Input() colSpan: number = 1;
+    @Input() get colSpan() {
+        return this.getColumnSpan();
+    }
+
+    set colSpan(colSpan: number) {
```

Comment:
You need to add the `ngAcceptInputType_colSpan` property in this component so that the Angular compiler knows that strings are valid values for this property as well.

Reply:
Thanks for that suggestion, allowed me to remove the isNaN changes from the dashboard.service.ts